rabbit-escape-engine/src/rabbitescape/engine/config/StandardConfigSchema.java

```diff
@@ -20,6 +20,18 @@ public static void setSchema( ConfigSchema definition )
             String.valueOf( false ),
             "Rabbit states are printed to System.out."
         );
+
+        definition.set(
+            CFG_WATER_DYN_CONTENTS_PER_PARTICLE,
+            String.valueOf( true ),
+            "Particle count is auto-reduced for struggling hardware."
+        );
+
+        definition.set(
+            CFG_WATER_CONTENTS_PER_PARTICLE,
+            String.valueOf( 4 ),
+            "Smaller values lead to more particles"
+        );
     }
```

Comment:
I think we need a new config version, since we're adding config properties.  The upgrade will probably be empty, but should set some default values if it is not OK for these values to be empty.

Alternatively, do we really need this to be configurable?  If not, I'd be fine with hard-coding these in the source code.

Reply:
I think I would like them to be configurable. I don't understand how the config versioning works. Can you explain? Or make the change?

Reply:
Why would you like them to be configurable?  Surely we just want to find the right values and set them?  Or do you see us shipping different builds with different config?  Or do you see the need for users to modify them?

Reply:
Some info on config upgrades: there is a list of how to upgrade from one version to another here: https://github.com/andybalaam/rabbit-escape/blob/master/rabbit-escape-engine/src/rabbitescape/engine/config/RealConfigUpgrades.java , and then the actual upgrades are in e.g. here: https://github.com/andybalaam/rabbit-escape/blob/master/rabbit-escape-engine/src/rabbitescape/engine/config/upgrades/UpgradeTo02LevelsCompletedByName.java

Essentially, you increase the config version number by 1, and make a new upgrader that converts from the previous version to the new one.

Bear in mind though that if you allow these settings to be missing and provide a default, we probably don't need a new version at all.  I don't remember, but I think that is true.